playerSDK/src/main/java/com/kaltura/playersdk/tracks/KTracksManager.java

```diff
@@ -40,6 +40,67 @@ public void switchTrack(TrackType trackType, int newIndex) {
     }
 
     @Override
+    public void switchTrackByBitrate(TrackType trackType, int prefaredBitrateKBit) {
+        Log.d(TAG, "switchTrackByBitrate : " + trackType.name() + " prefaredBitrateKBit : " + prefaredBitrateKBit);
+        if (TrackType.TEXT.equals(trackType)){
+            return;
+        }
+
+        int prevBitrate = 0;
+
+        List<TrackFormat> tracksList = null;
+        if (TrackType.AUDIO.equals(trackType)){
+            tracksList = getAudioTrackList();
+
+        } else if (TrackType.VIDEO.equals(trackType)){
+           tracksList =  getVideoTrackList();
+        } else {
+            //unsupported track type
+            return;
+        }
+
+        if (tracksList.size() == 1) {
+            Log.d(TAG, "Skip switchTrackByBitrate, tracksList.size() == 1");
+            return;
+        }
+        for (int trackIndex = 0 ; trackIndex <  tracksList.size() ; trackIndex++) {
```

Comment:
Why not just throwing all tracks to a TreeSet, sorted by Math.abs(track.bitrate-preferredBitrate) ?
Then you just take the first item on the set.